firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -201,6 +203,26 @@ public void onClose(Status status) {
                 handleWriteStreamClose(status);
               }
             });
+
+    networkReachabilityMonitor.onNetworkReachabilityChange(
+        (Reachability reachability) -> {
+          switch (reachability) {
+            case REACHABLE:
+              workerQueue.enqueueAndForget(
+                  () -> {
+                    disableNetwork();
```

Comment:
This is (probably) wrong. For one, the user could call FirebaseFirestore.enableNetwork/disableNetwork and this implementation would ignore that call and forcibly enable/disable the network upon a network change. Unless we *want* that to happen, we'll need to do something more intelligent here.

However, the point is that we want to reset all active connections upon a 'reachable' event. Ignoring the above, is this otherwise the right approach?

Reply:
As discussed, we're going to change this to ~ handleCredentialChange.